internal/librarian/generate.go

```diff
@@ -129,10 +130,11 @@ func runGenerate(ctx context.Context, cfg *config.Config) error {
 	}
 
 	image := deriveImage(ps)
-	containerConfig, err := docker.New(ctx, workRoot, image, cfg.SecretsProject, config)
+	secretsProvider, err := secrets.New(ctx, cfg.SecretsProject)
```

Comment:
Instead of initializing secrets.New here (and throughout the code), I think we want to call it inside config.New, then set the secret as a field on config

Reply:
Ok, I think this is what I was getting at in https://github.com/googleapis/librarian/pull/607#discussion_r2165036342. You're thinking of loading the secrets ahead of time in https://github.com/googleapis/librarian/blob/16c2b6c898bcab19c2c2a8076a10aa2d12db4153/internal/config/config.go#L261

I think this is a bigger refactor which can be done outside of this PR